### PR TITLE
Mobile navbar

### PR DESCRIPTION
**Pull request by oogxdd**
- Introduced mobile navbar

- Adapted existing pages content to look better on mobile (posts page, shop page, profile page)

- Introduced `/admin/site` route for the main settings tab. `/admin` route is now acting as the menu page if accessed from mobile, or if accessed from desktop - is redirecting to `/admin/site`

- Added settings/logout buttons to the profile page (on mobile)

- Fixed a bug where `@dir/ui`'s purged tailwind styles were overriding other tailwind styles (and it caused stuff like .flex overriding .md:flex) by whitelisting `@dir/core` in `@dir/ui`'s tailwind config. (Although I still think we duplicate tailwind styles at least in the development mode)

### CODE REVIEW

File: apps/web/app/[[...router]]/middleware.ts
Comment (dillonraphael) [inline apps/web/app/[[...router]]/middleware.ts:1]: Don't think this is needed?

File: packages/core/src/router.tsx [hunk 1]
```diff
@@ -529,172 +666,328 @@ export async function PageInit<T>({
...
+  router.addRoute(
+    "/admin",
+    async () => {
+      const headersList = headers()
+      const userAgent = headersList.get("user-agent")
+      const isMobile = /mobile/i.test(userAgent ?? "")
 
+      if (!isMobile) {
+        redirect("/admin/site")
+      }
```
Comment (dillonraphael) [inline packages/core/src/router.tsx:678]: Problem is this only works on mobile devices, and not on desktop and responsive

File: apps/web/app/[[...router]]/middleware.ts
Comment (oogxdd) [reply apps/web/app/[[...router]]/middleware.ts:1]: Yep

File: packages/core/src/router.tsx (hunk 1 above)
Comment (oogxdd) [reply packages/core/src/router.tsx]: Fixed

**Pull request merged**